### DISCUSSION TITLE
Reuse sisu-osgi to run Eclipse processes?

### DISCUSSION THREAD

**Discussion by lppedd** [eclipse-tycho/tycho#1787] [Q&A] [answered]
I have the necessity to launch an Eclipse process via a custom Maven plugin, to emulate a command line like:
```
my/custom/path/eclipse.exe -application org.eclipse.ant.core.antRunner -nosplash -buildfile build.xml
```
I'd like to avoid interacting with the bare JDK-provided process classes if possible, and was wondering, is using the Tycho `sisu-osgi` projects a good way to do it (like what the `eclipse-run` mojo does)?

**Comment by laeubi** [comment, marked as answer]
Yes `EclipseRunMojo` emulates something like this (and has a special constructor for extenders ... but I plan to improve that to better make it a component), but you can't "just" pass a eclipse install there. So if you really want to "execute eclipse" then `ProcessBuilder` is most likley what you want, because a seperate process is much more separated (e.g. has own process memory and alike)
